js-core/lesson1.md

```diff
@@ -4,198 +4,428 @@
 
 ```js
-countries.push('United States of America');
+function add (x, y) {
+  return x + y;
+}
 
-// .push can also be called with multiple parameters
-countries.push('United States of America', 'France');
+add(2, 3);
 ```
 
-> Homework: Learn about `.slice` and `.pop`
+Run this code, and hopefully you get the right answer!
+
+> **Exercise** Write a function called `multiply` that multiplies two numbers together. It should take two numbers as parameters and return the answer.
+
+# Arrays
+
+If you want to store a bunch of things, identifiers like `name1`, `name2`, `name3` and so on quickly become tiresome. **Arrays** are to the rescue!
+
+Imagine an array like a list or a collection of variables, but instead of each variable having its own name, only the array has a name. So how do you access a variable inside, then? How do you read and write? Using an *index*!
 
-There are many more useful array functions, which you will learn in lesson 2.
+An index is just a whole number, such as `0`, `1`, `2` and so forth. As in many areas of computing, indices in JavaScript arrays start with `0` --- so `0` is the index of the first element of an Array, `1` is the index of the second element, and so forth.
+
+The syntax is to list the values separated by commas, inside square brackets:
 
-## Loops
+```js
+[value1, value2, value3]
+```
 
-**TODO**
+You can store any type of value inside an array: strings, numbers, booleans. You can even store other arrays, or objects (which you will learn about next).
 
-### `for` loops
+## Try it out
 
-**TODO**
+Defining an array as a variable looks something like this:
+
+```js
+var animals = ["tiger", "puppy", "snake", "llama"];
+```
 
-#### Traversing an array
+Pretty neat! Now you have all the animals grouped together rather than having to define them separately. You can see how many animals there are by using `.length`:
 
-By now you might have noticed that `for` loops can be really useful for working with arrays.
+```js
+console.log(animals.length);
+```
 
-Imagine you want to print out every item in an array:
+If you want to refer to one specific member of the array, you can do so by using their "index number" like so.
 
 ```js
-var countries = ['Scotland', 'Germany', 'Syria'];
-var i;
+console.log(animals[1]);
+```
+
+Take a guess at which animal that will print out, and see if you're right.
+
+Spoiler alert, it will print out "puppy". That might seem weird, until you learn that JavaScript is a "0-index" language. What that means is that counting in JavaScript starts at 0. So `animals[0]`will print out "tiger". We don't make the rules...
+
+> **Exercise:** **Don't ask a mentor - Google it!**
+How would you `console.log()` the name of every animal in the array?
+If you're thinking a for loop, you're right. This is the perfect opportunity to use our newfound for loop skills for a higher purpose.
```

Comment:
`If you're thinking a for loop, you're right.` -> should we avoid telling them what to use, and give them more time to think?